Automattic/jetpack
# updates redirect package to accept all params

This PR updates the Redirect package to make it simpler and accept and pass through any parameter instead of having a controlled list.

This allows us to cover all situations where redirect links are created. It also makes it easier to add new params to the URL.

Also, this PR correctly passes the `unlinked` param inside the `query` as this is not used by the redirect server and we only need to send it to the final destination URL

#### Changes proposed in this Pull Request:

* Improves Jetpack Redirect package by accepting any parameter and making the docs better descriptive. 

#### Jetpack product discussion


p9dueE-36H-p2

#### Does this pull request change what data or activity we track or use?



#### Testing instructions:






* Connect  only at a site level
* Visit Appearance > Themes
* Wait a bit and you should see a JITM to activate backups
* hover the button on the JITM and make sure you see `query=unlinked=1` in the  URL
* Click the button
* Make sure `unlinked=1` is in the URL you go
* In the button to buy backup make sure there's `?unlinked=1` in the URL (image below)

![Captura de Tela 2021-06-25 às 15 41 59](https://user-images.githubusercontent.com/971483/123471319-faef5b80-d5cb-11eb-94dd-85bd816c32ad.png)
